### PR TITLE
Change readers to work on Stream of Bytes instead.

### PR DESCRIPTION
**Pull request by wishawa**
I've tried using `b2_upload_file` in my server and found working with Stream of Bytes more suitable than AsyncRead. Most server provides incoming request body as Stream of Bytes. Reqwest itself takes TryStream of Bytes to stream uploading. Wrapping Stream of Bytes into AsyncRead and back seems inefficient.

In this PR I change the readers utils to work on TryStream of Bytes instead of AsyncRead, and provide a function to convert the latter to the former.

Let me know what you think.

### CODE REVIEW

File: src/utils/readers.rs [hunk 1]
```diff
@@ -100,44 +97,50 @@ impl<R: AsyncRead> AsyncReadThrottled<R> {
...
-        let after_rmn = buf.remaining();
-        let read_amount = before_rmn - after_rmn;
-        let sleep_duration: f32 = (read_amount as f32) / *this.bandwidth;
-        this.sleep
-            .as_mut()
-            .reset(Instant::now() + Duration::from_secs_f32(sleep_duration));
-        Poll::Ready(Ok(()))
+        let res: Option<Result<Bytes, IoError>> = ready!(this.inner.poll_next(cx));
+        if let Some(Ok(bytes)) = &res {
+            let read_amount = bytes.len();
```
Comment (KongouDesu) [inline src/utils/readers.rs:111]: How much do we typically read per call here? I assume it depends on how much the inner returns, in which case, how many bytes does it typically return?
The sleep is based on how much we read and should be kept relatively short

Comment (wishawa) [reply src/utils/readers.rs:111]: Wrapping a file in FrameRead (as in `reader_to_stream`) produces 8k chunks. So delay should be < 100 ms for upload speed around 1 MBps.

I'm not sure about streams from network application and other sources.

Comment (KongouDesu) [reply src/utils/readers.rs:111]: Most network streams _should_ produce something like 4 or 8k chunk, so I think that should be fine.

**Comment by KongouDesu** [comment]
I think this sounds like a good idea. The utils are there to make it easier to work with, so they should just work without having to toss around more wrappers.

I'll get around to taking a closer look at this later today.

**Review by KongouDesu — APPROVED** [review]
Looks good to me.

**Pull request merged**